### PR TITLE
chat update and delete method with postman tests2

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
setup the Update and delete by ID chat methods in the controllers

Below are the Postman tests:-

1). Update by ID

image

2) Delete by ID

image

Now its delete so will not be found

image

continuation of The closed PR #49

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09  since i had made one step error there during rebase of PR #49 
I closed it and made a new one

Please see

File: src/modules/chat/chat.controller.ts [hunk 1]
```diff
@@ -54,14 +53,24 @@ export class ChatController {
     return await this.chatService.getChat(chatId);
   }
 
-  @Put('/update')
-  async updateChat(@Query('uid') uid, @Body() createChatDTO: CreateChatDTO) {
-    return await this.chatService.updateChat(uid, createChatDTO);
+  @Put('/:chatId')
+  async updateChat(
+    @Param('chatId') chatId: string,
+    @Body() createChatDTO: CreateChatDTO,
```
Comment (Abhishek-kumar09) [inline src/modules/chat/chat.controller.ts:59]: Why `createChatDto`, create a new DTO for chat updation.

File: src/modules/chat/chat.controller.ts [hunk 2]
```diff
@@ -54,14 +53,24 @@ export class ChatController {
...
 
   // Delete a Chat
-  @Delete('/delete')
-  async deleteChat(@Query('uid') uid) {
-    return await this.chatService.deleteChat(uid);
+  @Delete('/:chatId')
+  async deleteChat(@Param('chatId') chatId: string) {
+    const chat = await this.chatService.deleteChat(chatId);
+    if (!chat) throw new NotFoundException('Chat does not exist');
+    return chat;
```
Comment (Abhishek-kumar09) [inline src/modules/chat/chat.controller.ts:74]: Put these logics in service file, controller should be clean and only meant for handling request and responses

File: src/modules/chat/chat.controller.ts [hunk 3]
```diff
@@ -54,14 +53,24 @@ export class ChatController {
...
+  async updateChat(
+    @Param('chatId') chatId: string,
+    @Body() createChatDTO: CreateChatDTO,
+  ) {
+    console.log('chatId', chatId);
+    const chat = await this.chatService.updateChat(chatId, createChatDTO);
+
+    if (!chat) throw new NotFoundException('Chat does not exist!');
+
+    return chat;
```
Comment (Abhishek-kumar09) [inline src/modules/chat/chat.controller.ts:66]: These logics are redundant, already written in service file

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Create an chat architecture on paper or software of your choice, see how the chats should be organised, the datatypes of each of the fields, and how to distinguish between our written message and others.

Currently, there are problems with the chosen datatypes and some of the logic that are applied here.

File: src/modules/chat/chat.controller.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/modules/chat/chat.controller.ts:74]: ok

File: src/modules/chat/chat.controller.ts (hunk 1 above)
Comment (Devesh21700Kumar) [reply src/modules/chat/chat.controller.ts:59]: > Create an chat architecture on paper or software of your choice, see how the chats should be organised, the datatypes of each of the fields, and how to distinguish between our written message and others.
> 
> Currently, there are problems with the chosen datatypes and some of the logic that are applied here.

Thanks
Another question is that.. will we have a thread based chat structure? like in stack overflow?
or a normal linear chat structure where every user can create a chat in the course discussion?

**Comment by Abhishek-kumar09** [comment]
Hi @Devesh21700Kumar I think I merged this PR by mistake, sorry for inconvenience. Please apply the suggested changes and let me know if you have any doubt.

**Comment by Devesh21700Kumar** [comment]
> Create an chat architecture on paper or software of your choice, see how the chats should be organised, the datatypes of each of the fields, and how to distinguish between our written message and others.
> 
> Currently, there are problems with the chosen datatypes and some of the logic that are applied here.

image

Can u share your views on this workflow?

**Pull request merged**